Question Sharing documentation for Configuring Sharing in JSON (#10730)
PrairieLearn/PrairieLearn: docs/questionSharing.md

## Two ways to share: Publicly or through "Sharing sets"

> Questions can either be shared publicly, so that anyone can preview the questions and use them in their course, or you can share questions only to specific other courses using sharing sets.

Reviewer: ```suggestion
Questions can either be shared publicly, so that anyone can preview the questions and use them in their course, or shared only to specific other courses using sharing sets.
```

> Any question that is marked with `"sharePublicly": true` or `"shareSourcePublicly": true` will be considered and displayed as being published for free use under the [CC-BY-NC](https://www.creativecommons.org/licenses/by-nc/4.0/) license.

Reviewer: Nit:

```suggestion
Any question that is marked with `"sharePublicly": true` or `"shareSourcePublicly": true` will be considered and displayed as being published for free use under the [CC-BY-NC license](https://www.creativecommons.org/licenses/by-nc/4.0/).
```

## Sharing sets

> Access to shared questions which are not shared publicly is controlled through **sharing sets**. A sharing set is a named set of questions which you can share to another course all at once. The sharing set system exists so that course owners may differentially share different sets of their questions. For example, an instructor may want to share some questions only with other courses in their department, and other questions with anyone using PrairieLearn. Sharing sets are created by adding them to the [course configuration file](course.md#sharing-sets)

Reviewer: ```suggestion
Access to shared questions which are not shared publicly is controlled through **sharing sets**. A sharing set is a named set of questions which you can share to another course all at once. The sharing set system exists so that course owners may differentially share different sets of their questions. For example, an instructor may want to share some questions only with other courses in their department, and other questions with anyone using PrairieLearn. Sharing sets are created by adding them to the [`infoCourse.json` file](course.md#sharing-sets).
```

1. On your course admin page, visit the 'sharing' tab
2. Choose a sharing name for your course
3. Create a sharing set by adding it to your course's `infoCourse.json` file.
4. Have the instructor or the course you would like to share your question with visit the 'sharing' tab on their course admin page and provide you with their course's sharing token
5. Use the provided sharing token to add the other instructor's course as a consumer of the sharing set you created
6. Add the sharing set to the `sharingSets` list in `info.json` file of the questions you would like to share
7. The course you have shared the question with may now use it by referencing it in their assessments

## Steps to share a question publicly

**Note:** Once a question is publicly shared, the question cannot be renamed, deleted, or un-publicly shared because doing so could break the assessments of people who have used the shared question. Sharing the source code of a question may be un-done.

1. On your course admin page, visit the 'sharing' tab
2. Choose a sharing name for your course
3. Add `"sharePublicly": true` to the `info.json` file of questions you would like to share publicly

> 4. Optionally, add `"shareSourcePublicly": true` to the `info.json` file if you would like people to also be able to view and copy the source code of your question.

Reviewer: Nit: let's be consistent with either ending list items with periods, or not doing so. I'd be inclined to add periods everywhere.

Author: done

**Note:** Once a question is added to a sharing set, the question cannot be renamed, deleted, or removed from the sharing set because doing so could break the assessments of people who have used the shared question.

1. On your course admin page, visit the 'sharing' tab.
2. Choose a sharing name for your course.
3. Create a sharing set by adding it to your course's `infoCourse.json` file.
4. Have the instructor or the course you would like to share your question with visit the 'sharing' tab on their course admin page and provide you with their course's sharing token.
5. Use the provided sharing token to add the other instructor's course as a consumer of the sharing set you created.
6. Add the sharing set to the `sharingSets` list in `info.json` file of the questions you would like to share.
7. The course you have shared the question with may now use it by referencing it in their assessments.

## Steps to share a question publicly

**Note:** Once a question is publicly shared, the question cannot be renamed, deleted, or un-publicly shared because doing so could break the assessments of people who have used the shared question. Sharing the source code of a question may be un-done.

> 1. On your course admin page, visit the 'sharing' tab.

Reviewer: Extremely a nit, but on pretty much every other page but this one we use double quotes for things like this. Grep for `[a-zA-Z] '[a-zA-Z]` and `[a-zA-Z] "[a-zA-Z]` in `docs/` and compare.
